source/extensions/common/dynamic_forward_proxy/dns_cache.h

```diff
@@ -55,9 +63,37 @@ class DnsCacheResourceManager {
    * Returns the resource limit of pending requests to DNS.
    */
   virtual ResourceLimit& pendingRequests() PURE;
+
+  /**
+   * Returns the reference of stats for dns cache circuit breakers.
+   */
+  virtual DnsCacheCircuitBreakersStats& stats() PURE;
 };
 
 using DnsCacheResourceManagerPtr = std::unique_ptr<DnsCacheResourceManager>;
+using DnsCacheResourceManagerOptRef =
+    absl::optional<std::reference_wrapper<DnsCacheResourceManager>>;
+
+/**
+ * All DNS cache stats. @see stats_macros.h
+ */
+#define ALL_DNS_CACHE_STATS(COUNTER, GAUGE)                                                        \
```

Comment:
I don't see a reason to move this to the public interface. The overflow is a behavior of the cache implementation so it should be done internally to the cache.

Reply:
This is why we should have public interface to access DNS cache stats. It is required for this.
https://github.com/envoyproxy/envoy/pull/11028/files#diff-e3f6d903dd6231c359a06a057d7bdc7bR72

Reply:
Especially, we need this for this. https://github.com/envoyproxy/envoy/pull/11028/files#diff-bb768669231f226596558ac472b8d8b3R207

Reply:
Yes I understand why you did it, but I would prefer that you not leek the cache impl details into the filter. You should be doing the check and counter increment inside of a cache API.

Reply:
This needs to be reverted back to its previous location.